src/seq_analysis/orf.rs

```diff
@@ -225,4 +237,28 @@ mod tests {
         ];
         assert_eq!(expected, finder.find_all(sequence).collect::<Vec<Orf>>());
     }
+
+    #[test]
+    fn test_three_nested_and_offset_orfs() {
+        let finder = basic_finder();
+        let sequence = b"ATGGGGATGGGGGGATGGAAAAATAAGTAG";
+        let expected = vec![
+            Orf {
+                start: 14,
+                end: 26,
+                offset: 2,
+            },
+            Orf {
+                start: 0,
+                end: 30,
+                offset: 0,
+            },
+            Orf {
+                start: 6,
+                end: 30,
+                offset: 0,
+            },
+        ];
+        assert_eq!(expected, finder.find_all(sequence).collect::<Vec<Orf>>());
+    }
```

Comment:
Tests that nesting is detected and doesn't break scanning in other reading frames